### ISSUE TITLE
Dialog memory leaking

### ISSUE THREAD

**Issue by Exidel** [JetBrains/compose-jb#525]
Not sure if i'm using Dialog right or mb should create/use existed garbage collector but each time i open same Dialog window it consume memory and doesn't clear after dismiss.

https://user-images.githubusercontent.com/80458677/111909238-91f8c480-8a6d-11eb-9840-f93416b19f23.mp4
```

fun main() = Window(title = "TEST", size = IntSize(300, 300)) {

    MaterialTheme {

        Box(modifier = Modifier.padding(20.dp).fillMaxSize().border(10.dp, color = Color.Yellow).wrapContentSize())
            {


                val switch = remember { mutableStateOf(false)}

                Box(modifier = Modifier
                    .size( width = 100.dp, height =  100.dp)
                    .border(width = 5.dp, Color.Green)
                    .clickable(onClick = { switch.value = true }))



                if (switch.value) {

                    Dialog(
                        onDismissRequest = {switch.value = false},
                        properties = DialogProperties(
                            size = IntSize(300, 300),
                            title = "TEST TITLE",
                            resizable = false
                        )
                    ) {



                        Box(modifier = Modifier.fillMaxSize().border(width = 10.dp, Color.Red).wrapContentSize())
                        { Text(text = "TEST TEST TEST") }



                }

            }

        }

    }
}

```

**Comment by olonho** [comment]
Think we have a fix in Skiko.

**Comment by Exidel** [comment]
You mean i can avoid an issue by using Skiko render or am i miss the idea? On notebook i'm already use forced skiko render on PC - common one (don't know which one is basic) - both show same result (diffirent memory consuming on window creation but still). I thought that there are still references from shared variables but no, even without any refered variables and content with simplest code it's still the same.

**Comment by Exidel** [comment]
I see this is already fixed in 0.4.0-build177

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-5471">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
